clients/unshaded/src/main/java/tachyon/client/file/FileInStream.java

```diff
@@ -215,7 +221,7 @@ private void checkAndAdvanceBlockInStream() throws IOException {
         try {
           // TODO(calvin): Specify the location to be local.
           mCurrentCacheStream =
-              mContext.getTachyonBlockStore().getOutStream(currentBlockId, -1, null);
+              mContext.getTachyonBlockStore().getOutStream(currentBlockId, -1, mLocation.getHost());
```

Comment:
I don't think you want this to be the host being read from right? Since presumably the data is already there? Instead probably want to only write to local.